flake.nix

```
@@ -32,13 +32,20 @@
           legacyPackages = devshell;
           devShell = devshell.fromTOML ./devshell.toml;
         };
+      flakeTOML = inputs: path: eachSystem (system:
+        let pkgs = nixpkgs.legacyPackages.${system}.extend self.overlay;
+        in { 
+          devShell = pkgs.devshell.fromTOML path;
```

```suggestion
      flakeTOML = nixpkgs: path: eachSystem (system:
        let devshell = import ./. { pkgs = nixpkgs.legacyPackages.${system}; }
        in { 
          devShell = devshell.fromTOML path;
```
Like this, no new instance of nixpkgs is being created. Each instance adds quite a big overhead to nix evaluation. Composition over inheritance :)

---

I liked the idea but it didn't work :(

When run `nix develop it says...

> error: package "devshell.cli" not found

Changed PR to draft until we solve this

Should I revert?

---

Reverted ☹